framework/src/utils/PetscSupport.C

```diff
@@ -632,6 +632,26 @@ getCommonPetscKeys()
     "-snes_ls -snes_max_it -snes_rtol -snes_type -sub_ksp_type -sub_pc_type", "", true);
 }
 
+void
+setSinglePetscOption(const char * name, const char * value)
```

Comment:
I think it would be better if this function took `const std::string &` instead of `const char *`and then we called PETSc with `name.c_str()` and `value.c_str()`.